src/db.c

```diff
@@ -2210,3 +2210,30 @@ int xreadGetKeys(struct redisCommand *cmd, robj **argv, int argc, getKeysResult
     result->numkeys = num;
     return num;
 }
+
+/* Helper function to extract keys from the SET command, which may have
+ * a read flag if the GET argument is passed in. */
+int setGetKeys(struct redisCommand *cmd, robj **argv, int argc, getKeysResult *result) {
+    keyReference *keys;
+    UNUSED(cmd);
+
+    keys = getKeysPrepareResult(result, 1);
+    keys[0].pos = 1; /* We always know the position */
+    keys[0].flags = CMD_KEY_UPDATE;
+
+    for (int i = 1; i < argc; i++) {
```

Comment:
first, we need to start the search from 3 (not 1), since the value or the key-name can be the word "get" (i.e. `set foo get`).
secondly, some of the flags have argument (e.g. EX), so to search there it is invalid.
although these arguments are always numbers, it's a bit odd to look into them.

i.e. imagine the case of `set foo bar ex get`.
it's an invalid command, but not sure it should be flagged with ACCESS and rejected by ACL.
on the other hand, the alternative is to either clone the entire command parsing, or share code with t_string.c
so maybe i'm ok with your approach and it's limitations.

Reply:
right, i forget it. changed it to 3.

>on the other hand, the alternative is to either clone the entire command parsing, or share code with t_string.c
>so maybe i'm ok with your approach and it's limitations.
i am keeping this one for now, and we can wait for more suggestions. I would be more inclined to `clone the entire command parsing` part

Reply:
looks like other commands have the same problem (e.g. georadiusGetKeys), so i guess that's fine.